src/rpc/server.cpp

```diff
@@ -511,6 +516,8 @@ UniValue CRPCTable::execute(const std::string &strMethod, const UniValue &params
                             params.size(),
                             helpMsg));
             } else {
+                int64_t nElapsedExecutionTime = (GetTimeMicros() - nExecutionStartTime);
+                LogPrint("rpc", "ThreadRPCServer completed execution of method in %.2fms [%.2fs]", nElapsedExecutionTime * 0.001, nElapsedExecutionTime * 0.000001);
                 return pcmd->actor(params, false);
```

Comment:
```suggestion
                auto result = pcmd->actor(params, false);
                int64_t nElapsedExecutionTime = (GetTimeMicros() - nExecutionStartTime);
                LogPrint("rpc", "ThreadRPCServer completed method=%s in %.2fms", SanitizeString(strMethod), nElapsedExecutionTime * 0.001);
                return result;
```
There are three independent suggestions here.

You have to run the RPC before you stop the timer, otherwise, the reported time won't include the actual RPC execution.

Reporting milliseconds is sufficient.

It would be good to log the method string again because multiple RPCs can run in parallel, so we should be able to tell which one took a particular amount of time.